engine/src/flutter/impeller/renderer/backend/vulkan/swapchain/khr/khr_swapchain_vk.cc

```diff
@@ -56,8 +61,19 @@ std::unique_ptr<Surface> KHRSwapchainVK::AcquireNextDrawable() {
     return std::move(result.surface);
   }
 
+  constexpr const size_t kMaxResizeAttempts = 3u;
+  if (resize_retry_count == kMaxResizeAttempts) {
+    VALIDATION_LOG << "Attempted to resize the swapchain" << kMaxResizeAttempts
+                   << " time unsuccessfully. This platform likely doesn't "
+                      "support returning the current swapchain extents and "
+                      "must recreate the swapchain using the actual size.";
+    return nullptr;
+  }
+
   TRACE_EVENT0("impeller", "RecreateSwapchain");
 
+  size_ = impl_->GetCurrentUnderlyingSurfaceSize().value_or(size_);
```

Comment:
I think this code is wrong, and that is resulting in the broken goldens.

Reply:
Possibly: not handling transform? i.e. needs w/h to be swapped?

Reply:
okay, no i remember this - the Vulkan surface query will give out of date information compared to the rotation state in Android.  That is, when you rotate the phone, the query will return the unrotated size for several additional frames after the embedding tells the engine that it has rotated.

On Android we cannot use this information to determine screen size or rotation.

Reply:
Thanks for the tip. I worked around it using an `ifdef` but this will cause issues for Android users who have no way of explicitly setting the surface size. Filed a followup https://github.com/flutter/flutter/issues/163070 that I will fix after this lands.